include/hipCPU/hip/detail/event.hpp

```diff
@@ -29,33 +29,40 @@
 #define HIPCPU_EVENT_HPP
 
 #include <atomic>
+#include <chrono>
 
 namespace hipcpu {
 
 class event
 {
 public:
   event()
-  : _is_recorded{false}
+  : _record_timestamp{0}
   {}
 
   void wait() const noexcept
   {
-    do {} while (!_is_recorded);
+    do {} while (!_record_timestamp);
   }
 
   void mark_as_finished()
   {
-    _is_recorded = true;
+    auto now = std::chrono::steady_clock::now().time_since_epoch();
+    _record_timestamp = std::chrono::duration_cast<std::chrono::nanoseconds>(now).count();
   }
 
   bool is_complete() const
   {
-    return _is_recorded;
+    return _record_timestamp != 0;
+  }
+
+  uint64_t timestamp_ns() const
+  {
+    return _record_timestamp;
   }
 
 private:
-  std::atomic<bool> _is_recorded;
+  std::atomic<uint64_t> _record_timestamp;
```

Comment:
While it probably won't be a problem in practice, I'm a bit skeptical about using a timestamp to determine whether an event is complete or not. Building a system where things can go horribly wrong if the system time is set to an unfortunate date such that `_record_timestamp` is set to 0 in `mark_as_finished()` doesn't sound like a good idea to me.
Would it be too ugly to keep `_is_recorderd` and just have and additional `_record_timestamp`?

Reply:
I tried avoiding an additional atomic store by keeping state in a single variable. I have changed the behavior to never store a zero, even if `steady_clock` behaves unexpectedly (an implementation where `now()` is constant-zero would b conformant I think).